src/services/user-group/services/courseScopeLists.js

```diff
@@ -0,0 +1,63 @@
+const { /* ScopePermissionService, */ ScopeListService } = require('../../helpers/scopePermissions');
+const { courseModel } = require('../model');
+
+module.exports = function setup() {
+	const app = this;
+
+	ScopeListService.initialize(app, '/users/:scopeId/courses', async (user, permissions, params) => {
+		let filter = 'active';
+		let substitution = 'false';
+		if (params.query.filter && ['active', 'archived', 'all'].includes(params.query.filter)) {
+			({ filter } = params.query);
+		}
+		if (params.query.substitution && ['true', 'false', 'all'].includes(params.query.substitution)) {
+			({ substitution } = params.query);
+		}
+
+		const userQuery = { $or: [] };
+		if (['false', 'all'].includes(substitution)) {
+			userQuery.$or.push(
+				{ userIds: user._id },
+				{ teacherIds: user._id },
+			);
+		}
+		if (['true', 'all'].includes(substitution)) userQuery.$or.push({ substitutionIds: user._id });
+
+		let untilQuery = {};
+		if (filter === 'active') {
+			untilQuery = {
+				$or: [
+					{ untilDate: { $exists: false } },
+					{ untilDate: { $gte: Date.now() } },
+				],
+			};
+		}
+		if (filter === 'archived') {
+			untilQuery = { untilDate: { $lt: Date.now() } };
+		}
+
+		if (params.query.count === 'true') {
+			const courseCount = await courseModel.count({
+				$and: [
+					userQuery,
+					untilQuery,
+				],
+			});
```

Comment:
https://mongoosejs.com/docs/promises.html#queries-are-not-promises

Reply:
if the query is `await`ed only once, I don't see the problem.